packages/shared/src/hooks/post/useBlockPostPanel.ts

```diff
@@ -75,10 +75,11 @@ export const useBlockPostPanel = (
   const client = useQueryClient();
   const { user } = useAuthContext();
   const { checkHasCompleted, completeAction } = useActions();
-  const key = generateStorageKey(
-    StorageTopic.Post,
-    `block:${post?.id}`,
-    user?.id,
+  const key = useMemo(
+    () => [
+      generateStorageKey(StorageTopic.Post, `block:${post?.id}`, user?.id),
```

Comment:
Our function always returns an array right? Probably just a matter of type?

Reply:
generateStorageKey returns a string, it's used for storage parts, but mixed use in this specific implementation as a react query key as well.

Reply:
Probably better to use wrapped in `generateQueryKey`?

Reply:
Hmm ideally yes, this edge-case has local state as well which uses storage key, so to keep them sync I choose to leave it.

Ideally secondary task should be: move everything to use generateQueryKey (also the hardcoded string arrays now)

Reply:
But keep the storage key and when used in `useQuery` just wrap it in `generateQueryKey`. This way if react-query down the line changes query keys to what ever we just need to change `generateQueryKey` implementation vs looking for all array initializations.

Reply:
But then let's just do this right away, I just wanted to avoid all this work now 😅

Reply:
I don't think we need to do all, we touch this part so let's change it, but that works as well🙏

Reply:
Actually moving this to `queryKey` now will introduce a key change, which no longer matches with the stored one, as our generate function will always add `annonymous` as extra prop.

Reply:
Don't want to block it by additional work if we can do it down the line.

Reply:
Changed all together 😂